src/sentry/integrations/github/client.py

```diff
@@ -104,73 +104,74 @@ def get_tree(self, repo_full_name: str, tree_sha: str) -> JSONData:
+        contents into the cache.
         """
         trees: Dict[str, RepoTree] = {}
         cache_key = f"githubtrees:repositories:{cache_key}:{gh_org}"
-        repo_key = "githubtrees:repo"
-        cached_repositories = cache.get(cache_key, [])
-        if not cached_repositories:
+        repositories = cache.get(cache_key)
+        if not repositories:
             # Simply removing unnecessary fields from the response
             repositories = [
                 {"full_name": repo["full_name"], "default_branch": repo["default_branch"]}
                 for repo in self.get_repositories(fetch_max_pages=True)
             ]
-            # XXX: In order to speed up this function we will need to parallelize this
-            # Use ThreadPoolExecutor; see src/sentry/utils/snuba.py#L358
-            for repo_info in repositories:
-                try:
-                    full_name: str = repo_info["full_name"]
-                    branch = repo_info["default_branch"]
-                    files = self.get_repo_files(full_name, branch)
-                    repo = Repo(full_name, branch)
-                    trees[full_name] = RepoTree(repo, files)
-                    cache.set(f"{repo_key}:{full_name}", trees[full_name], cache_seconds)
-                except Exception:
-                    # Catching the exception ensures that we can make progress with the rest
-                    # of the repositories
-                    logger.exception(f"We have failed to fetch the tree for {repo_info}.")
             cache.set(cache_key, repositories, cache_seconds)
             next_time = datetime.now() + timedelta(seconds=cache_seconds)
             logger.info(f"Caching trees for {gh_org} org until {next_time}.")
-        else:
-            try:
-                for repo_info in cached_repositories:
-                    repo_tree = cache.get(f"{repo_key}:{repo_info['full_name']}")
-                    # This assertion will help clear the cache off dictionaries
-                    assert type(repo_tree) == RepoTree
-                    trees[repo_info["full_name"]] = repo_tree
-
-                logger.info(f"Using cached trees for {gh_org}.")
-            except AssertionError:
-                # Reset the control cache in order to repopulate
-                cache.delete(cache_key)
-                logger.exception(f"We reset the cache for {cache_key}.")
+
+        # XXX: In order to speed up this function we will need to parallelize this
+        # Use ThreadPoolExecutor; see src/sentry/utils/snuba.py#L358
+        try:
+            for repo_info in repositories:
+                full_name = repo_info["full_name"]
+                branch = repo_info["default_branch"]
+                repo_files = self.get_cached_repo_files(full_name, branch)
+                trees[full_name] = RepoTree(Repo(full_name, branch), repo_files)
+            logger.info(f"Using cached trees for {gh_org}.")
```

Comment:
Generally we include variable content for log in the `extra` parameter as our logging infra struggles with high cardinality